docs/user-guide/grpc.md

```diff
@@ -4,133 +4,209 @@
----
-apiVersion: ambassador/v0
-kind: Mapping
-name: grpc_mapping
-grpc: true
-prefix: /helloworld.Greeter/
-rewrite: /helloworld.Greeter/
-service: grpc-greet
+      ---
+      apiVersion: ambassador/v0
+      kind: Mapping
+      name: grpc_py_mapping
+      grpc: True
+      prefix: /helloworld.Greeter/
+      rewrite: /helloworld.Greeter/
+      service: grpc-example
 ```
 
-Note the `grpc: true` line -- this is the necessary magic when mapping a gRPC service. Also note that you'll need `prefix` and `rewrite` the same here, since the gRPC service needs the package and service to be in the request to do the right thing.
-
-## Deploying `Hello World`
+Note the `grpc: true` line —- this is what tells envoy to use HTTP/2 so the request can communicate with the backend service. Also note that you'll need `prefix` and `rewrite` the same here, since the gRPC service needs the package and service to be in the request to do the right thing.
 
-To deploy and map `Hello World`, we can use the following YAML:
+### Deploying to kubernetes
+`grpc_example.yaml`
 
 ```yaml
 ---
 apiVersion: v1
 kind: Service
 metadata:
   labels:
-    service: grpc-greet
-  name: grpc-greet
+    service: grpc-example
+  name: grpc-example
   annotations:
     getambassador.io/config: |
       ---
       apiVersion: ambassador/v0
       kind: Mapping
-      name: grpc_mapping
-      grpc: true
+      name: grpc_py_mapping
+      grpc: True
       prefix: /helloworld.Greeter/
       rewrite: /helloworld.Greeter/
-      service: grpc-greet
+      service: grpc-example
 spec:
   type: ClusterIP
   ports:
-  - port: 80
-    name: grpc-greet
+  - name: grpc-greet
+    port: 80
     targetPort: grpc-api
   selector:
-    service: grpc-greet
+    service: grpc-example
 ---
 apiVersion: extensions/v1beta1
 kind: Deployment
 metadata:
-  name: grpc-greet
+  name: grpc-example
 spec:
   replicas: 1
   template:
     metadata:
       labels:
-        service: grpc-greet
+        service: grpc-example
     spec:
       containers:
-      - name: grpc-greet
-        image: enm10k/grpc-hello-world
+      - name: grpc-example
+        image: <docker_hub>/grpc_example
         ports:
         - name: grpc-api
-          containerPort: 9999
-        env:
-          - name: PORT
-            value: "9999"
-        command:
-          - greeter_server
+          containerPort: 50051
       restartPolicy: Always
 ```
 
-(We tell the gRPC service to run on port 9999, then map the container's port 80 inbound to simplify the `Mapping`. There's no magic behind these port numbers: anything will work as long as you're consistent in when mapping everything.)
+Once adding the mapping configuration to the service with an annotation, the rest of the kubernetes deployment YAML file is pretty straightforward. We need to identify the container image to use, expose the `containerPort` to listen on the same port the docker container is listening on, and map the service port (80) to the container port (50051).
```

Comment:
Once the mapping configuration has been added to the service ...

(Awkward sentence). Also Kubernetes should be capitalized, as does Docker.